assets/js/payment-methods-demo/express-payment/apple-pay/apple-pay-express.js

```diff
@@ -0,0 +1,380 @@
+/**
+ * Internal dependencies
+ */
+import {
+	DEFAULT_STRIPE_EVENT_HANDLERS,
+	PAYMENT_METHOD_NAME,
+} from './constants';
+import {
+	getBillingData,
+	getPaymentMethodData,
+	normalizeShippingAddressForCheckout,
+	normalizeShippingOptions,
+	normalizeLineItems,
+} from './normalize';
+import {
+	getStripeServerData,
+	getPaymentRequest,
+	updatePaymentRequest,
+	canDoApplePay,
+	getTotalPaymentItem,
+	stripePromise,
+} from '../../stripe-utils';
+
+/**
+ * External dependencies
+ */
+import { useRef, useState, useEffect } from '@wordpress/element';
+import {
+	Elements,
+	PaymentRequestButtonElement,
+	useStripe,
+} from '@stripe/react-stripe-js';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * @typedef {import('../../stripe-utils/type-defs').Stripe} Stripe
+ * @typedef {import('../../stripe-utils/type-defs').StripePaymentRequest} StripePaymentRequest
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').RegisteredPaymentMethodProps} RegisteredPaymentMethodProps
+ */
+
+/**
+ * @typedef {Object} WithStripe
+ *
+ * @property {Stripe} [stripe] Stripe api (might not be present)
+ */
+
+/**
+ * @typedef {RegisteredPaymentMethodProps & WithStripe} StripeRegisteredPaymentMethodProps
+ */
+
+// @todo, note it should be trivial to implement cross browser payment api
+// handling here:
+// - ApplePay in Safari
+// - ChromePay in Chrome
+// - not supported for other browsers yet (but if assuming stripe implements the
+//   official PaymentRequest API in their library this should enable support!).
+
+/**
+ * AppleExpressComponent
+ *
+ * @param {StripeRegisteredPaymentMethodProps} props Incoming props
+ */
+const ApplePayExpressComponent = ( {
+	paymentStatus,
+	shippingData,
+	billing,
+	eventRegistration,
+	onSubmit,
+	activePaymentMethod,
+	setActivePaymentMethod,
+} ) => {
+	/**
+	 * @type {[ StripePaymentRequest|null, function( StripePaymentRequest ):StripePaymentRequest|null]}
+	 */
+	// @ts-ignore
```

Comment:
Is this temporary?

Reply:
Yes. I have a personal todo (not an issue) to review all @ts-ignore usages and see about fixing. I just don't want to waste time on fixing things that is because I can't figure out how to properly document a particular type.